### PR TITLE
Updating federation up scripts to work in non e2e setup

### PR DESCRIPTION
**Pull request by nikhiljindal**
Ref: https://github.com/kubernetes/kubernetes.github.io/pull/656

Updating the federation up scripts so that they work as per steps in https://github.com/kubernetes/kubernetes.github.io/pull/656.

Changes are:
- Updating the default namespace to be "federation" instead of "federation-e2e"
- Updated the kubeconfig context to be named "federation-cluster" instead of "federated-context"
- Fixing federation-up so that FEDERATION_IMAGE_TAG is set even when federation-up is run without running `e2e.go --up`. e2e-up.sh sets it here: https://github.com/kubernetes/kubernetes/blob/6a388d4a0d9e7fc2549ab89c97bf29ce53982e39/hack/e2e-internal/e2e-up.sh#L44.
- Adding a "missingkey=zero" option to template parser. Without this, the parser adds `"<no value>"` at the place of an env var that is not set. With this change, it instead replaces it with the corresponding zero value (for ex "" for strings). This is required for the FEDERATION_DNS_PROVIDER_CONFIG env var.

cc @kubernetes/sig-cluster-federation @colhom @mml

### CODE REVIEW

**Comment by quinton-hoole** [comment]
LGTM!  Thanks @nikhiljindal

**Comment by colhom** [comment]
@nikhiljindal you're missing the `test/` side of things which consumes these constants.

* `test/e2e/{framework/test_context,federation-apiserver}.go` contain hard-coded references to old `federated-cluster` kubecontext name

* `test/e2e/framework/{util.framework}.go` have hard-coded references to old `federation-e2e` namespace name.

**Comment by colhom** [comment]
@quinton-hoole please remove lgtm label. This PR is not complete and will break the tests.

**Comment by quinton-hoole** [comment]
Good point @colhom - removed LGTM until fixed.

**Comment by nikhiljindal** [comment]
Thanks @quinton-hoole and @colhom 
Updated the code to fix those references

File: federation/cluster/federation-up.sh [hunk 1]
```diff
@@ -22,4 +22,8 @@ KUBE_ROOT=$(readlink -m $(dirname "${BASH_SOURCE}")/../../)
 
 . ${KUBE_ROOT}/federation/cluster/common.sh
 
+if [[ -f "${KUBE_ROOT}/federation/manifests/federated-image.tag" ]]; then
```
Comment (colhom) [inline federation/cluster/federation-up.sh]: This has been something I've been meaning to do. We actually be very specific here:

```sh
tagfile="${KUBE_ROOT}/federation/manifests/federated-image.tag"
if [[ ! -f "$tagfile" && "${FEDERATION:-}" == "true ]];then
    echo "FATAL: FEDERATION=true, but tagfile ${tagfile} does not exist"
    exit 1
fi

**Comment by colhom** [comment]
@nikhiljindal there is one more old namespace reference in `./hack/e2e-internal/e2e-status.sh`

File: federation/cluster/federation-up.sh (hunk 1 above)
Comment (nikhiljindal) [inline federation/cluster/federation-up.sh]: Done.
We do not check if FEDERATION=true in federation-up.sh

**Comment by nikhiljindal** [comment]
Thanks @colhom 
Fixed and pushed a new version.
PTAL.

**Comment by nikhiljindal** [comment]
Found a missed reference while testing. Fixed it now.
Also added a test to verify that cluster is marked ready by the controller.

Verified that the test passes!

**Comment by nikhiljindal** [comment]
Adding back LGTM

**Comment by nikhiljindal** [comment]
Fixed verify-govet

**Comment by nikhiljindal** [comment]
Merge both is not showing this PR in the merge queue.

Trying to run the e2e test again, to see if that triggers the merge bot to look at this again.

@k8s-bot e2e test this issue: #IGNORE

**Pull request merged**